### PR TITLE
fix: handle maxQty negative in discount code

### LINKED ISSUE #7474: Handle the maxQuantity being negative
**Describe the bug**

Presently maxQuantity for discount codes is 100000.
Handle the maxQuantity being negative, i.e. -1
If it is negative, server should not limit the discount code count.

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #7474 

#### Short description of what this resolves:
Handle maxQty negative in discount code by removing the limitation for tickets to be less than maxqty when it is negative.

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@daretobedifferent18 can you kindly review if these are the changes we are looking for? I tried to solve it according to my understanding :sweat_smile: Also @iamareebjamal With the test failing I checked the logs it's the same error I am getting that I posed on gitter `'xyz' has no attribute 'data_key'`. Can you please help me out? Thanks :v:

File: app/api/schema/discount_codes.py [hunk 1]
```diff
@@ -94,11 +94,12 @@ def validate_quantity(self, data, original_data):
         DiscountCodeSchemaEvent.quantity_validation_helper(data)
 
         if data.get('tickets_number') and data.get('max_quantity'):
-            if data['tickets_number'] < data['max_quantity']:
-                raise UnprocessableEntityError(
-                    {'pointer': '/data/attributes/tickets-number'},
-                    "tickets-number should be greater than max-quantity",
-                )
+            if data['max_quantity'] >= 0:
+                if data['tickets_number'] < data['max_quantity']:
```
Comment (iamareebjamal) [inline app/api/schema/discount_codes.py:98]: and

File: app/api/schema/discount_codes.py [hunk 2]
```diff
@@ -166,11 +167,12 @@ def validate_quantity(self, data, original_data):
         DiscountCodeSchemaTicket.quantity_validation_helper(data)
 
         if data.get('tickets_number') and data.get('max_quantity'):
-            if data['tickets_number'] < data['max_quantity']:
-                raise UnprocessableEntityError(
-                    {'pointer': '/data/attributes/tickets-number'},
-                    "tickets-number should be greater than max-quantity",
-                )
+            if data['max_quantity'] >= 0:
+                if data['tickets_number'] < data['max_quantity']:
```
Comment (iamareebjamal) [inline app/api/schema/discount_codes.py:171]: and

**Comment by iamareebjamal** [comment]
Strange that tests are only failing on your PRs

**Comment by iamareebjamal** [comment]
Other changes touching this are still left

File: app/api/schema/discount_codes.py (hunk 1 above)
Comment (Tlazypanda) [reply app/api/schema/discount_codes.py:98]: @iamareebjamal I am not sure I follow ? :sweat_smile:

File: app/api/schema/discount_codes.py (hunk 2 above)
Comment (Tlazypanda) [reply app/api/schema/discount_codes.py:171]: @iamareebjamal I am not sure I follow ? sweat_smile

Comment (iamareebjamal) [reply app/api/schema/discount_codes.py:171]: Can be made one statement using and condition

Comment (Tlazypanda) [reply app/api/schema/discount_codes.py:171]: ohh sure sure making that change right away!

**Comment by Tlazypanda** [comment]
@iamareebjamal can you review now and let me know if there are some other changes to be made? :sweat_smile:

**Comment by Tlazypanda** [comment]
@iamareebjamal I found one another place where we are dependent on max_quantity so i modified the condition to bypass the negative values but not sure if this is how we want it handled can you have a look? :sweat_smile:

**Pull request merged**